PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -84,6 +91,20 @@ public class PocketSource: Source {
     public func object<T: NSManagedObject>(id: NSManagedObjectID) -> T? {
         space.object(with: id)
     }
+
+    private func observeNetworkStatus() {
+        networkMonitor.start(queue: .main)
+        networkMonitor.updateHandler = { path in
+            switch path.status {
+            case .unsatisfied, .requiresConnection:
+                self.syncQ.isSuspended = true
+            case .satisfied:
+                self.syncQ.isSuspended = false
+            @unknown default:
+                self.syncQ.isSuspended = false
+            }
+        }
+    }
```

Comment:
This bit here is really the crux of the whole thing. Since all of our operations are executed in order via `self.syncQ`, when the user goes offline we simply pause the queue. Local changes will take effect immediately while the remote operations wait in the queue.
When the user goes back online, we resume the queue and the operations are executed on the server in order.

Reply:
I'm not sure whether we should pull down any changes _before_ executing the local changes or if that will be necessary?

Reply:
I think we can resort to a K.I.S.S approach and send our changes up _to_ the server, and let the next "sync" with data _from_ the server update the local entries as necessary. TL;DR - Local is the source-of-truth until we request that the server tell us otherwise.